keps/sig-node/5419-pod-level-resources-in-place-resize/README.md

```diff
@@ -0,0 +1,904 @@
+
+## Motivation
+
+The ability to specify resource requirements at the pod level, as introduced by KEP-2837, offers a more simplified and efficient way to manage resource consumption for multi-container pods. However, without the ability to resize these pod-level allocations in-place, operators would still be forced to recreate pods to adjust their overall resource footprint, negating some of the benefits of dynamic resource management.
+
+Enabling in-place resizing for pod-level resources complements KEP-2837 by allowing:
+
+* Dynamic Adjustment of Aggregate Resources: Operators can scale the total resources
+  available to a pod up or down based on changing workload demands, without
+  disrupting the services running within.
+
+* Improved Resource Utilization: By dynamically adjusting pod-level limits, clusters
+  can respond more efficiently to fluctuations in aggregate resource usage,
+  potentially leading to better packing and reduced over-provisioning.
+
+* Reduced Operational Overhead: Eliminating the need for pod recreation for resource
+  adjustments simplifies operational tasks and reduces downtime or disruption for
+  applications.
+
+### Goals
+
+This proposal aims to:
+
+1. Extend the In-Place Pod Resize (IPPR) functionality to support dynamic adjustments of pod-level CPU, Memory, and Hugepages resources.
+2. Ensure compatibility and proper interaction between pod-level IPPR and existing container-level IPPR mechanisms.
+3. Provide clear mechanisms for tracking and reporting the actual allocated pod-level resources in PodStatus
+
+## Proposal
+### Notes/Constraints/Caveats
+
+1. cgroupv1 has been moved to maintenance mode since Kubernetes version 1.31.
+   Hence this feature will be supported only on cgroupv2. Kubelet will fail to
+   admit pods with pod-level resources on nodes with cgroupv1.
+
+2. Currently, in-place pod resizing for pod-level resources will be guarded behind a
+   separate feature gate, InPlacePodLevelResourcesVerticalScaling, which will be
+   available in alpha from v1.34. InPlacePodLevelResourcesVerticalScaling is an
+   **opt-in** feature, and will have no effect on existing deployments.
+
+3. This feature relies on the PodLevelResources feature gate being enabled.
+
+## Design Details
```

Comment:
Most of the existing metrics added by KEP-1287 can likely be extended to include changes to Pod level resources. The `kubelet_container_requested_resizes_total` metric seems to count changes at the container level, but would it be better to add a new metric like `kubelet_pod_requested_resizes_total` to specifically count changes to Pod level resources?
https://github.com/kubernetes/enhancements/blob/ef7e11d088086afd84d26c9249a4ca480df2d05a/keps/sig-node/1287-in-place-update-pod-resources/README.md#kubelet_container_requested_resizes_total

Reply:
Yes, I would like to add these as a part of Beta. Hyperlinking it in this KEP